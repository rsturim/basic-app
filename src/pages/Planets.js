import React from 'react';

import { Link } from 'react-router-dom';

import PlanetList from '../components/PlanetList';

import Nav from '../components/Nav';
import Header from '../components/Header';

export default function Planets() {
    return (
        <div className="container">
            <Header>Planet List</Header>
            <Nav item="/planets" />
            <PlanetList />
        </div>
    );
}
