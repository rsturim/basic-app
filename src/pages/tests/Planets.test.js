import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import { mount, shallow } from 'enzyme';

import toJson from 'enzyme-to-json';

import Planets from '../Planets';

describe('The Planets page', () => {
    test('should do shallow rendering', () => {
        // const wrapper = mount(<Planets />);

        const wrapper = shallow(<Planets />);
        expect(wrapper.find('Header').length).toBe(1);

        // console.log(wrapper.debug());
    });

    test('should do a Full DOM Render', () => {
        const wrapper = mount(
            <MemoryRouter>
                <Planets />
            </MemoryRouter>,
        );

        expect(wrapper.find('Header').length).toBe(1);
        expect(wrapper.find('Header').text()).toBe('Planet List');

        // console.log(wrapper.debug());
    });

    test('render a dom snapshot for testing', () => {
        const wrapper = shallow(<Planets />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

/**
<MemoryRouter>
A <Router> that keeps the history of your "URL" in memory (does not read or write to the address bar).
Useful in tests and non-browser environments like React Native.

https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/MemoryRouter.md
*/
