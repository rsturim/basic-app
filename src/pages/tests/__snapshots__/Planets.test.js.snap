// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Planets page render a dom snapshot for testing 1`] = `
<div
  className="container"
>
  <Header>
    Planet List
  </Header>
  <Nav
    item="/planets"
  />
  <PlanetList />
</div>
`;
